pkg/typesystem/typesystem.go

```diff
@@ -7,53 +7,23 @@ import (
 	openfgapb "go.buf.build/openfga/go/openfga/api/openfga/v1"
 )
 
-type SchemaVersion int32
-
-const (
-	SchemaVersionUnspecified SchemaVersion = 0
-	SchemaVersion1_0         SchemaVersion = 1
-	SchemaVersion1_1         SchemaVersion = 2
-)
-
-var (
-	ErrInvalidSchemaVersion = errors.New("invalid schema version")
-)
-
-func NewSchemaVersion(s string) (SchemaVersion, error) {
-	switch s {
-	case "", "1.0":
-		return SchemaVersion1_0, nil
-	case "1.1":
-		return SchemaVersion1_1, nil
-	default:
-		return SchemaVersionUnspecified, ErrInvalidSchemaVersion
-	}
-}
-
-func (v SchemaVersion) String() string {
-	switch v {
-	case SchemaVersion1_0:
-		return "1.0"
-	case SchemaVersion1_1:
-		return "1.1"
-	default:
-		return "unspecified"
-	}
-}
+var ErrDuplicateTypes = errors.New("an authorization model cannot contain duplicate types")
 
 type TypeSystem struct {
-	Version         SchemaVersion
+	Version         string
```

Comment:
This and `TypeDefinitions` don't need to be publicly exported. We want to encourage usage of the methods themselves instead of indexing into the fields of the TypeSystem.

Reply:
I am not 100% sure I agree, but let's try it out. Done. Thanks for the comment!